### PR TITLE
Map sidecar `roletoken` config to SIA config

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

- refactor config package
- new SIA interface
  - ENV: `TOKEN_EXPIRY`
  - args: `token-expiry`
- default values
  - token refresh: `30m`
  - token expiry: `0` (implies requesting to Athenz ZTS server WITHOUT request parameter)

## Type of change

- [ ] Bug fix
- [x] New feature
- [x] Refactoring (no functional changes, no api changes)
- [ ] Non-code changes (update documentation, pipeline, etc.)

### Flags

- [x] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code

### CODE REVIEW

File: pkg/config/default.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+	// DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES may be overwritten with go build option (e.g. "-X identity.DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES=5")
+	DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES_RAW = "5"
+	DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES     = 5
+
+	DEFAULT_ENDPOINT                     string
+	DEFAULT_ROLE_AUTH_HEADER             = "Athenz-Role-Auth"
+	DEFAULT_DNS_SUFFIX                   = "athenz.cloud"
+	DEFAULT_ROLE_CERT_FILENAME_DELIMITER = ":role."
+	DEFAULT_INTERMEDIATE_CERT_BUNDLE     string
+)
```
Comment (mlajkim) [inline pkg/config/default.go:44]: @WindzCUHK 
1. why the are upper case snake case? should be CamelCase?
2. why are they inside of the `var`, when they should not be modified as default values? (`const`?)

File: pkg/identity/identity.go [hunk 2]
```diff
@@ -475,7 +409,7 @@ func (h *identityHandler) GetToken(certPEM, keyPEM []byte) (roletokens [](*RoleT
 	// The intermediate certificates may be different between each ZTS.
 	// Therefore, ZTS Client for PostRoleCertificateRequest must share the same endpoint as PostInstanceRegisterInformation/PostInstanceRefreshInformation
 	roleClient := zts.NewClient(h.config.Endpoint, t)
-	expireTimeMs := int32(DEFAULT_TOKEN_EXPIRY_TIME_INT * 60)
+	expireTimeMs := int32(config.DEFAULT_TOKEN_EXPIRY_TIME_INT * 60)
```
Comment (mlajkim) [inline pkg/identity/identity.go:412]: fix it.

**Review by mlajkim — APPROVED** [review]
OK. Fix the pipeline though!

File: pkg/config/default.go (hunk 1 above)
Comment (WindzCUHK) [reply pkg/config/default.go:44]: > why the are upper case snake case? should be CamelCase?

All of them are constants.

> why are they inside of the var, when they should not be modified as default values? (const?)

To inject value during `go build` (see Makefile), ONLY variable declaration can be overwritten.

File: pkg/identity/identity.go (hunk 2 above)
Comment (WindzCUHK) [reply pkg/identity/identity.go:412]: This logic is removed in later PR. Just ignore is OK.

File: pkg/config/default.go (hunk 1 above)
Comment (WindzCUHK) [reply pkg/config/default.go:44]: @mlajkim 
related discussion for your information: https://github.com/golang/go/issues/22706

**Pull request merged**